File: hooks/webhook_receiver.py
from flask import Flask, request, abort
app = Flask(__name__)
WEBHOOK_SECRET = "mysecret"

@app.route("/webhook", methods=["POST"])
def webhook():
    if request.headers.get("X-Webhook-Secret") != WEBHOOK_SECRET:
        abort(401)
    data = request.json
    print("Received:", data)
    return "", 200

if __name__ == "__main__":
    app.run(port=8080)

File: hooks/webhook_receiver.go
package main
import (
  "encoding/json"
  "fmt"
  "net/http"
)

const secret = "mysecret"

func webhookHandler(w http.ResponseWriter, r *http.Request) {
  if r.Header.Get("X-Webhook-Secret") != secret {
    http.Error(w, "unauthorized", http.StatusUnauthorized)
    return
  }
  var data map[string]interface{}
  json.NewDecoder(r.Body).Decode(&data)
  fmt.Println("Received:", data)
  w.WriteHeader(http.StatusOK)
}

func main() {
  http.HandleFunc("/webhook", webhookHandler)
  http.ListenAndServe(":8080", nil)
}